sdk/resourcemanager/azure-resourcemanager-keyvault/src/main/java/com/azure/resourcemanager/keyvault/implementation/SecretImpl.java

```diff
@@ -76,24 +98,20 @@ public boolean managed() {
         return vault
             .secretClient()
             .listPropertiesOfSecretVersions(name())
-            .flatMap(p -> {
-                if (p.isEnabled()) {
-                    return vault.secretClient().getSecret(p.getName(), p.getVersion());
-                } else {
-                    return Mono.just(new KeyVaultSecret(p.getName(), null).setProperties(p));
-                }
-            })
```

Comment:
@xccc-msft 
Now that I've removed the per-item GET request, and added the `enabled` method, I believe I can let customer handle this case. Customer should be able to check `enabled` and decide whether to do a `getValue` or not.

Reply:
The change looks good. One question is for the secret version. When the customer want to get value for enabled secret, I think `.getSecret(name, version)` should have version param passed too. Not sure if I missed any part, but it seems it alwarys pass NULL as per your changes.